core/server/master/src/main/java/alluxio/master/file/InodeSyncStreamPendingPathCollection.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * PendingPath in InodeSyncStream type.
+ */
+public interface InodeSyncStreamPendingPathCollection {
```

Comment:
Instead of dong this, I'm think whether it would be better if we replace ConcurrentLinkedQueue<AlluxioURI> with ConcurrentLinkedDeque<AlluxioURI> and adds the new element to the head or the tail of the deque based on the InodeSyncStreamPendingPathCollectionType? @tcrain @jiacheliu3 any thoughts here?

Reply:
Yea using a ConcurrentLinkedDeque and adding the head/tail based on the configuration sounds much cleaner.

Reply:
Hello @adol001 we haven't seen your reply on this comment. :)

Reply:
@elega  Replaced ConcurrentLinkedQueue with ConcurrentLinkedDeque.

Reply:
thanks. my point is that we can just replace the ConcurrentLinkedQueue https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java#L274-L275 with ConcurrentLinkedDeque and decide if you should do poll or pollLast based on the configuration. If you have concerns over the poll() method being called multiple times in InodeSyncStream in the future, you can add a helper function for that. This is optional to me though as there's only one usage currently.

This is the only comment I have and apart from this your PR looks good to me. Thanks for the great work!

Reply:
Fixed, thank you for your review.